element.go

```diff
@@ -729,6 +754,78 @@ func readExplicit(buffer *dicomio.Decoder, tag dicomtag.Tag) (string, uint32) {
 	return vr, vl
 }
 
+// readNativeFrames reads Native frames from a Decoder based on already parsed pixel information
+// that should be available in parsedData (elements like NumberOfFrames, rows, columns, etc)
+func readNativeFrames(d *dicomio.Decoder, parsedData *DataSet) (pixelData *PixelDataInfo, bytesRead int, err error) {
+	image := PixelDataInfo{
+		Encapsulated: false,
+	}
+
+	rows, err := parsedData.FindElementByTag(dicomtag.Rows)
+	if err != nil {
+		return nil, 0, err
+	}
+	cols, err := parsedData.FindElementByTag(dicomtag.Columns)
+	if err != nil {
+		return nil, 0, err
+	}
+	nof, err := parsedData.FindElementByTag(dicomtag.NumberOfFrames)
+	nFrames := 0
+	if err != nil {
+		//TODO(suyash): explicitly check for not found error, currently assume single frame image
+		nFrames = 1
+	}
+	if nFrames == 0 {
+		nFrames, err = strconv.Atoi(nof.MustGetString()) // odd that number of frames is encoded as a string...
+		if err != nil {
+			dicomlog.Vprintf(1, "ERROR converting nof")
+			return nil, 0, err
+		}
+	}
+
+	b, err := parsedData.FindElementByTag(dicomtag.BitsAllocated)
+	if err != nil {
+		dicomlog.Vprintf(1, "ERROR finding bits allocated.")
+		return nil, 0, err
+	}
+	bitsAllocated := int(b.MustGetUInt16())
+	image.BitsPerSample = bitsAllocated
+
+	s, err := parsedData.FindElementByTag(dicomtag.SamplesPerPixel)
+	if err != nil {
+		dicomlog.Vprintf(1, "ERROR finding samples per pixel")
+	}
+	samplesPerPixel := int(s.MustGetUInt16())
+
+	pixelsPerFrame := int(rows.MustGetUInt16()) * int(cols.MustGetUInt16())
+
+	dicomlog.Vprintf(1, "Image size: %d x %d", rows.MustGetUInt16(), cols.MustGetUInt16())
+	dicomlog.Vprintf(1, "Pixels Per Frame: %d", pixelsPerFrame)
+	dicomlog.Vprintf(1, "Number of frames %d", nFrames)
+
+	// Parse the pixels:
```

Comment:
consider breaking this out into a more abstract function that takes a `io.Reader` and `nFrames`, `pixelsPerFrame`, etc and returns a `[][][]int` parsed pixel data. 

May also be good to add a `pixeldata` package or an `image` package that has utility functions and type definitions for specifically parsing pixel data / images. This `element.go` is getting a bit large, and likely should live in a package (perhaps a `parse` package?)

Reply:
larger refactors should probably happen in a separate dedicated diff thoguh